server/athenian/api/precompute/discover_accounts.py

```diff
@@ -28,5 +38,39 @@ async def main(context: PrecomputeContext, args: argparse.Namespace) -> List[int
         if state.finished_date is None:
             log.warning("Skipped account %d because the progress is not 100%%", account)
             continue
-        result.append(account)
-    return result
+
+        discovered.append((account, precomputed))
+
+    if args.partition:
+        result: Dict[str, List[int]] = {"precomputed": [], "fresh": []}
+        for precomputed, group_accounts in groupby(
+            sorted(discovered, key=itemgetter(1)), key=itemgetter(1),
+        ):
+            result_field = "precomputed" if precomputed else "fresh"
+            result[result_field] = [account for account, _ in group_accounts]
```

Comment:
I know that this code is correct and works, but here is a much simpler to comprehend alternative:
```suggestion
        result: Dict[str, List[int]] = {"precomputed": [], "fresh": []}
        for account, precomputed in discovered:
            result["precomputed" if precomputed else "fresh"].append(account)
        for accounts in result.values():
            accounts.sort()
```

Reply:
intention here was to preserve accounts order in both cases.
Rewritten as you suggested.
Anyway now we can also remove `ORDER BY` from query ?

Reply:
We can remove it but then we should sort client-side if `partition = False`.

The goal of this sorting was to start precomputing from the oldest account to the newest, so that if we crash on a new one, all the rest are precomputed. If we `partition = True`, we don't care about the order anymore, but it's still nice for determinism.

Reply:
I forgot that we sort the result now, so yeah, let's remove.

Reply:
order by removed